sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -97,6 +97,49 @@ kernel_pid_t gnrc_ipv6_init(void)
     return gnrc_ipv6_pid;
 }
 
+static void _dispatch_insert_ext_headers(gnrc_pktsnip_t *pkt)
+{
+    DEBUG("ipv6: dispatch insertion of extension headers\n");
+    uint8_t ext_demux[] = { PROTNUM_IPV6_EXT_HOPOPT
+                          , PROTNUM_IPV6_EXT_RH
+                          , PROTNUM_IPV6_EXT_FRAG
+                          , PROTNUM_IPV6_EXT_ESP
+                          , PROTNUM_IPV6_EXT_AH
+                          , PROTNUM_IPV6_EXT_DST
+                          , PROTNUM_IPV6_EXT_MOB
+                          };
+    bool encapsulate = false;
+
+    for ( size_t i = 0; i < sizeof(ext_demux); ++i ) {
+        gnrc_netreg_entry_t *call = gnrc_netreg_lookup(GNRC_NETAPI_MSG_TYPE_SND, ext_demux[i]);
+        msg_t msg, rcv;
+        msg.type = ext_demux[i];
+
+        while (call) {
+            if (   ext_demux[i] == PROTNUM_IPV6_EXT_HOPOPT
+                || ext_demux[i] == PROTNUM_IPV6_EXT_RH ) {
+                encapsulate = true;
+            }
+            msg.content.ptr = pkt;
+            msg_send_receive(&msg, &rcv, call->target.pid);
```

Comment:
I'm not sure if you use `NETAPI_SND` here, because above you use `NETAPI_SND` in a weird context, but `NETAPI_SND` uses asynchronous API, so this messes up things.